tests/kernel/semaphore/semaphore_error_cases/src/main.c

```diff
@@ -0,0 +1,331 @@
+/*
+ * Copyright (c) 2020 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+#include <irq_offload.h>
+
+#define STACK_SIZE (512 + CONFIG_TEST_EXTRA_STACKSIZE)
+#define THREAD_TEST_PRIORITY 0
+
+static struct k_thread tdata;
+static K_THREAD_STACK_DEFINE(tstack, STACK_SIZE);
+
+static struct k_sem sema;
+
+/* action after self-defined fatal handler. */
+static ZTEST_BMEM volatile bool valid_fault;
+static ZTEST_BMEM volatile bool assert_in_isr_has_done;
+
+
+static inline void set_fault_valid(bool valid)
+{
+	valid_fault = valid;
+}
+
+void k_sys_fatal_error_handler(unsigned int reason, const z_arch_esf_t *pEsf)
```

Comment:
didn't you guys introduce a generic framework for handling fatal errors in tests? can we use this here?

Reply:
@enjiamai  What do you think of?

Reply:
@andrewboie Sure, we write it this way because the common fatal handler PR doesn't get merged yet, It's block by me( in change request status and need examples and tests), I don't what to block their test cases.  That definitely better than writing such a code snippet every where. I will make this PR ready as possible today. Then we can modified our lots of our code to use that.